tests/e2e/event_handler/test_header_serializer.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from requests import Request
+
+from tests.e2e.utils import data_fetcher
+
+
+@pytest.fixture
+def alb_basic_listener_endpoint(infrastructure: dict) -> str:
+    dns_name = infrastructure.get("ALBDnsName")
+    port = infrastructure.get("ALBBasicListenerPort", "")
+    return f"http://{dns_name}:{port}"
+
+
+@pytest.fixture
+def alb_multi_value_header_listener_endpoint(infrastructure: dict) -> str:
+    dns_name = infrastructure.get("ALBDnsName")
+    port = infrastructure.get("ALBMultiValueHeaderListenerPort", "")
+    return f"http://{dns_name}:{port}"
+
+
+@pytest.fixture
+def apigw_rest_endpoint(infrastructure: dict) -> str:
+    return infrastructure.get("APIGatewayRestUrl", "")
+
+
+@pytest.fixture
+def apigw_http_endpoint(infrastructure: dict) -> str:
+    return infrastructure.get("APIGatewayHTTPUrl", "")
+
+
+@pytest.fixture
+def lambda_function_url_endpoint(infrastructure: dict) -> str:
+    return infrastructure.get("LambdaFunctionUrl", "")
+
+
+def test_alb_headers_serializer(alb_basic_listener_endpoint):
+    # GIVEN
+    url = f"{alb_basic_listener_endpoint}/todos"
+
+    # WHEN
+    response = data_fetcher.get_http_response(Request(method="GET", url=url))
+
+    # THEN
+    assert response.status_code == 200
+    assert response.content == b"Hello world"
+    assert response.headers["content-type"] == "text/plain"
+
+    # Only the last header for key "Foo" should be set
+    assert response.headers["Foo"] == "zbr"
+
+    # Only the last cookie should be set
+    assert "MonsterCookie" in response.cookies.keys()
+    assert "CookieMonster" not in response.cookies.keys()
```

Comment:
pr: Pass input to function as POST data instead of hardcoding it in the handler.

Reply:
we can do a cleanup in another PR - `POST` and a more realistic cookie in the docs.